### DISCUSSION TITLE
Can you put a menu or sub-menu inside an If else statement or void function?

### DISCUSSION THREAD

**Discussion by Makima-sama** [forntoh/LcdMenu#129] [Q&A] [answered]
So in the machine I was going to create I have 3 buttons(up,down,enter), 2x16LCD, 2 float sensor.
What I want to happen is to create a menu in which the menu will only show specific items or choices depends on the current state of the float sensor.
[The code that I made is already long and most of it is not related in this question so I will just put the important part]

In order to run specific codes based on the current state of float sensor I used if-else statement like this:
`
void fsState(){
  int floatSwitch1State = !digitalRead(fSwitch1Pin); //I use INPUT_PULLUP in void setup 
  int floatSwitch2State = !digitalRead(fSwitch2Pin);

  if (floatSwitch1State == HIGH && floatSwitch2State == LOW) {
  //Storage 1 not empty
 // THIS IS WHERE I WANT TO PUT THE SUBMENU or menu
  } 
  else if (floatSwitch1State == LOW && floatSwitch2State == HIGH) {
  //Storage 2 not Empty
 // THIS IS WHERE I WANT TO PUT THE SUBMENU or menu
  } 
  else if (floatSwitch1State == HIGH && floatSwitch2State == HIGH) {
  //Both Storage not empty
  } 
  else {
  //Both Storage empty
  }

  delay(100);
}
`
ps. This code does not work when you put submenu inside.
So basically what I did and I want to happen is use  **ITEM_COMMAND("float sensor", fsState)** it will run the fsState Void function and show the submenu or menu based on the current state of the float switch.
For example:
Float switch on Storage 1 is High, it will show on the LCD.
1. Dispense water (ITEM_COMMAND)
2. Return to mainmenu (ITEM_COMMAND)
when swtich on storage 2 is high.
1.Pump to plants (ITEM_COMMAND)
2. return to mainmenu (ITEM_COMMAND)

If there are other ways to create a menu that shows a sub-menu depends on the current state of something please kindly tell.

**Comment by forntoh** [comment, marked as answer]
It's not possible to build a menu dynamically (on runtime) with this menu system, you can update already existing menus but can't create.

However, your logic seems very simple, you can just update the text of the ITEM_COMMAND and like set a bool var to true which you can check in the callback function and decide the appropriate action to take.

**Reply by Makima-sama** [reply]
Hello thanks for answering that was quite fast.

Can you please give me an example on how to set bool var in the ITEM_COMMAND?
I read the tutorial on LCD dlcumentation specifically the "Command and Toggle"
https://lcdmenu.forntoh.dev/examples/command-and-toggle
I'm not really familiar with this type of code and I slightly get the method theh did on Toggle but that is not really I want to use.
